### DISCUSSION TITLE
Scrub inline dates?

### DISCUSSION THREAD

**Discussion by glen-84** [VerifyTests/Verify#959] [Q&A] [answered]
Is it possible to scrub inline date-times in a JSON string (`VerifyJson(string)`), similar to `ScrubInlineGuids`, or would you have to use `ScrubLines` along with a regex?

The incrementing `DateTimeOffset_x` values are causing issues, so we just want a fixed `{Scrubbed}` value.

Ideally it would reuse existing date matching logic.

**Comment by SimonCropp** [comment]
i have considered this a few times, but always decided no to due to the complexity of defining what a date is in any string. especially when you consider the wacky date format in the US.

but you should be able to write your own with ScrubLinesWithReplace https://github.com/VerifyTests/Verify/blob/main/docs/serializer-settings.md

**Reply by glen-84** [reply]
Hmm. I'm a bit confused.

Something _was_ "scrubbing" the date/times in these strings, that's why we got `DateTimeOffset_1`, etc. Not the same as just `{Scrubbed}`, but still they were being matched and replaced somewhere _when given a string_.

Does Verify deserialize the JSON string first? That might explain it.

Either way, the request would be to optionally use a single replacement string like `{Scrubbed}` instead of the default `DateTimeOffset_n` naming.

For now, we're using `ScrubLinesWithReplace`, but it's probably doing unnecessary work.

Thanks.

**Reply by SimonCropp** [reply]
i still dont understand why u want “scrubbed” instead of “date_x”

**Reply by glen-84** [reply]
You can ignore this request for now.

I had an issue where some dates were sometimes equal, and sometimes not, depending on the speed of execution. However, I've started to utilize `FakeTimeProvider` in more places now, so this should no longer be an issue.

Thanks.

**Reply by SimonCropp** [reply]
> some dates were sometimes equal, and sometimes not, depending on the speed of execution

that is usually due to multiple uses if DateTime.Now within the same context. which is often indicative of other bugs. for example if daylight savings happens between two DateTime.Now  calls

**Reply by glen-84** [reply]
I'm only using UTC, so DST shouldn't be an issue.

What was happening is that when I register a `User`, I also create a `RoleAssignment`, and both the user and the role assignment have a `CreatedAt` date/time. Sometimes the role assignment would be created at exactly the same time as the user, and sometimes not.

I've updated the code to set the `CreatedAt` date/time of the role assignment equal to the `CreatedAt` date/time of the user, and the user's `CreatedAt` date/time is also set to a static date/time from the `FakeTimeProvider`. I'm not sure if it's ideal (since having slightly different times seems more accurate), but it's probably okay for now.

So basically, any time that you create more than one entity with a date/time, the times may match (or differ slightly) if you don't force the same time with a `FakeTimeProvider` or set the dates equal to one another. I still think that it's more realistic for these dates to vary, and therefore the `DateTimeOffset_x` naming is not always appropriate.

**Reply by SimonCropp** [reply]
can u try 22.11.0 and https://github.com/VerifyTests/Verify/blob/main/docs/dates.md#disabledatecounting

**Reply by glen-84** [reply]
It doesn't seem to be working: https://github.com/glen-84/verify-959.

**Reply by SimonCropp** [reply, marked as answer]
can u try 22.11.1

**Reply by glen-84** [reply]
It works. 🎉

Thanks Simon.
